sdk/src/wallet/operations/output_consolidation.rs

```diff
@@ -153,41 +182,36 @@ where
                 .await?;
             if !is_locked_output && should_consolidate_output {
                 outputs_to_consolidate.push(output_data.clone());
+
+                // Keep track of inputs with native tokens.
+                if let Some(nt) = &output_data.output.native_token() {
+                    native_token_inputs
+                        .entry(*nt.token_id())
+                        .or_insert_with(HashSet::new)
+                        .insert(output_data.output_id);
+                }
             }
         }
 
+        // Remove outputs if they have a native token, <= minimum amount and there are no other outputs with the same
+        // native token.
+        outputs_to_consolidate.retain(|output_data| {
+            output_data.output.native_token().as_ref().map_or(true, |nt| {
+                if output_data.output.amount() <= output_data.output.minimum_amount(storage_score_parameters) {
```

Comment:
Not sure to understand this

Reply:
Outputs created in the genesis snapshot can have a lower amount than min amount (or if the min amount increases at some point)

Reply:
Added a comment

Reply:
Ah, forgot about that, thanks